index.js

```diff
@@ -645,22 +663,51 @@
     }
   };
 
-  //  commonTypes :: [[Type]] -> [Type]
-  var commonTypes = function(typeses) {
-    if (isEmpty(typeses)) return [];
-
-    return chain(typeses[0], function(t) {
-      var common = true;
-      for (var idx = 1; idx < typeses.length; idx += 1) {
-        common = false;
-        for (var idx2 = 0; idx2 < typeses[idx].length; idx2 += 1) {
-          if (equalTypes(t, typeses[idx][idx2])) {
-            common = true;
-            break;
-          }
-        }
+  //  commonTypes :: ([[Type]], Boolean) -> [Type]
+  var commonTypes = function(typeses, loose) {
```

Comment:
Sure.

Let's say we have `values :: [0, 0.5, 1]`.

This might give `typeses :: [[Number, Integer], [Number], [Number, Integer]]`.

`commonTypes(typeses, false)` would then evaluate to `[Number]`.

Now let's discuss the `loose` parameter.

Let's say we have `values :: ['foo', false]`.

This might give `typeses :: [[String], [Boolean]]`.

`commonTypes(typeses, false)` would then evaluate to `[]`, indicating a type error.

We'd like to display `['foo', false] :: Array ???` rather than `['foo', false] :: ???`. The `loose` parameter allows us to do so. We set `loose` to `true` when determining validity; `false` when generating error messages.